File: golangSdk/golandSdk/apiInterface.go
package ugcapi

import (
	"strconv"
)

type ImageBucketListParam struct {
	Region     string //required
	BucketName string //option  Docker镜像仓库名称，不填返回所有的镜像仓库
	BucketType string //option  默认All  All：用户所有可见的仓库 User：用户个人创建的仓库 Share：第三方公开镜像
	Orderby    string //option  Default 默认排序
	Limit      int    //option query num 默认20
	Offset     int    //option  默认为0
	PublicKey  string //required  用户的 公钥
	PrivateKey string //required  用户的 私钥
}

//获取可用的Docker镜像仓库列表，包括公有和私有的
func GetBucketImagesList(request ImageBucketListParam) (result string, err error) {
	var paramMap map[string]string = make(map[string]string)

	paramMap["Region"] = request.Region
	if len(request.BucketName) != 0 {
		paramMap["BucketName"] = request.BucketName
	}

	if len(request.BucketType) != 0 {
		paramMap["BucketType"] = request.BucketType
	}

	if len(request.Orderby) != 0 {
		paramMap["OrderBy"] = request.Orderby
	}

	if request.Limit != 0 {
		paramMap["Limit"] = strconv.Itoa(request.Limit)
	}

	if request.Offset != 0 {
		paramMap["Offset"] = strconv.Itoa(request.Offset)
	}

	paramMap["Action"] = "GetImageBucketList"
	paramMap["PublicKey"] = request.PublicKey

	signature := Vefsig(request.PrivateKey, paramMap)
	paramMap["Signature"] = signature
	url := GenerateUrl(COMMON_API_URL, paramMap)

	return HttpGet(url)
}

type DockerImageListReqest struct {
	Region     string //required
	BucketName string //required  Docker镜像仓库名称
	Orderby    string //option  Default: 默认排序
	Limit      int    //option query num 默认20
	Offset     int    //option 默认0
	PublicKey  string //required  用户的 公钥
	PrivateKey string //required  用户的 私钥
}

//获取可用的Docker镜像列表，包括公有和私有的
func GetDockerImagesList(request DockerImageListReqest) (rsp string, err error) {
	paramMap := make(map[string]string)
	paramMap["Region"] = request.Region
	paramMap["BucketName"] = request.BucketName

	if len(request.Orderby) != 0 {
		paramMap["OrderBy"] = request.Orderby
	}

	if request.Limit != 0 {
		paramMap["Limit"] = strconv.Itoa(request.Limit)
	}

	if request.Offset != 0 {
		paramMap["Offset"] = strconv.Itoa(request.Offset)
	}

	paramMap["Action"] = "GetDockerImageList"
	paramMap["PublicKey"] = request.PublicKey

	signature := Vefsig(request.PrivateKey, paramMap)
	paramMap["Signature"] = signature
	url := GenerateUrl(COMMON_API_URL, paramMap)

	return HttpGet(url)
}

type AllImagesRequest struct {
	Region     string //required
	PublicKey  string //required  用户的 公钥
	PrivateKey string //required  用户的 私钥
}

//获取所有可用的Docker镜像名称，包括公有和私有的
func GetAllImage(request AllImagesRequest) (rsp string, err error) {
	paramMap := make(map[string]string)
	paramMap["Region"] = request.Region
	paramMap["Action"] = "GetAllImages"
	paramMap["PublicKey"] = request.PublicKey

	signature := Vefsig(request.PrivateKey, paramMap)
	paramMap["Signature"] = signature
	url := GenerateUrl(COMMON_API_URL, paramMap)

	return HttpGet(url)
}

type CreateDockerImageBucketRequest struct {
	Region     string //required
	BucketName string //required
	PublicKey  string //required  用户的 公钥
	PrivateKey string //required  用户的 私钥
}

//创建Docker镜像bucket
func CreateDockerImageBucket(request CreateDockerImageBucketRequest) (rsp string, err error) {
	paramMap := make(map[string]string)
	paramMap["Region"] = request.Region
	paramMap["Action"] = "CreateDockerImageBucket"
	paramMap["BucketName"] = request.BucketName
	paramMap["PublicKey"] = request.PublicKey

	signature := Vefsig(request.PrivateKey, paramMap)
	paramMap["Signature"] = signature
	url := GenerateUrl(COMMON_API_URL, paramMap)

	return HttpGet(url)
}

type GetTaskListRequest struct {
	Region     string //required
	TaskId     string //option 任务ID 不填返回所有Id的任务
	TaskName   string //option 任务名称  不填返回所有名称的任务
	TaskType   string //option  同步任务"Sync"  异步任务 "Async"  默认"All"
	State      string //option  Running：运行中  Success：成功  Fail：失败  默认为 All:所有
	Orderby    string //option 排序
	Limit      int    //optio   默认为20
	Offset     int    //option  默认为0
	PublicKey  string //required  用户的 公钥
	PrivateKey string //required  用户的 私钥
}

// 获取提交的任务列表
func GetTaskList(request GetTaskListRequest) (rsp string, err error) {
	paramMap := make(map[string]string)
	paramMap["Region"] = request.Region
	paramMap["Action"] = "GetTaskList"
	paramMap["PublicKey"] = request.PublicKey

	if len(request.TaskId) != 0 {
		paramMap["TaskId"] = request.TaskId
	}

	if len(request.TaskName) != 0 {
		paramMap["TaskName"] = request.TaskName
	}

	if len(request.TaskType) != 0 {
		paramMap["TaskType"] = request.TaskType
	}

	if len(request.State) != 0 {
		paramMap["State"] = request.State
	}

	if len(request.Orderby) != 0 {
		paramMap["OrderBy"] = request.Orderby
	}

	if request.Limit != 0 {
		paramMap["Limit"] = strconv.Itoa(request.Limit)
	}

	if request.Offset != 0 {
		paramMap["Offset"] = strconv.Itoa(request.Offset)
	}

	signature := Vefsig(request.PrivateKey, paramMap)
	paramMap["Signature"] = signature
	url := GenerateUrl(COMMON_API_URL, paramMap)

	return HttpGet(url)

}

type GetTaskDetailRequest struct {
	Region     string //required
	TaskId     string //required
	PublicKey  string //required  用户的 公钥
	PrivateKey string //required  用户的 私钥
}

// 查看任务详情
func GetTaskDetail(request GetTaskDetailRequest) (rsp string, err error) {
	paramMap := make(map[string]string)
	paramMap["Region"] = request.Region
	paramMap["Action"] = "GetTaskDetail"
	paramMap["TaskId"] = request.TaskId
	paramMap["PublicKey"] = request.PublicKey

	signature := Vefsig(request.PrivateKey, paramMap)
	paramMap["Signature"] = signature
	url := GenerateUrl(COMMON_API_URL, paramMap)

	return HttpGet(url)
}

type GetAccessTokenRequest struct {
	Region     string //required
	ExpireIn   int    //option  token 有效时间 单位秒  默认 7200 秒
	GrantType  string //option  默认为task
	PublicKey  string //required  用户的 公钥
	PrivateKey string //required  用户的 私钥
}

//提交任务接口使用AccessToken做身份验证，此接口用于获取AccessToken
func GetAccessToken(request GetAccessTokenRequest) (rsp string, err error) {
	paramMap := make(map[string]string)
	paramMap["Region"] = request.Region
	paramMap["Action"] = "GetAccessToken"
	if len(request.GrantType) != 0 {
		paramMap["GrantType"] = request.GrantType
	}
	if request.ExpireIn != 0 {
		paramMap["ExpireIn"] = strconv.Itoa(request.ExpireIn)
	}

	paramMap["PublicKey"] = request.PublicKey

	signature := Vefsig(request.PrivateKey, paramMap)
	paramMap["Signature"] = signature
	url := GenerateUrl(COMMON_API_URL, paramMap)

	return HttpGet(url)

}

type SubmitTaskRequest struct {
	Region         string //required
	ImageName      string //required
	Cmd            string //option   命令行参数
	OutputDir      string //option   结果输出路径
	OutputFilename string //option 结果保存文件名
	TaskType       string //option "Sync"  同步 "Async"  异步
	Timeout        int    //option  默认10s  单位为妙
	TaskName       string //option   任务名称
	AccessToken    string //required
}

/* 提交任务

返回值
提交失败
{
	"Action" : "SubmitTaskResponse",
	"RetCode" : -1,
	"Message" : “All Fail”
}


同步任务提交成功

返回任务输出的二进制tar文件，目录结构如下，子文件仅在不为空的时候返回

Output-> 返回的tar格式二进制文件
Stdout  任务的标准输出
Stderr   任务的标准错误输出
OutputDir\OutputFileName  任务设置的文件输出

异步任务提交成功
{
	"Action" : "SubmitTaskResponse",
	"RetCode" : 0,
	"Message" : “”,
	"TaskId" : “123455”,
}

*/

func SubmitTask(request SubmitTaskRequest, data string) (rsp string, err error) {
	paramMap := make(map[string]string)
	paramMap["Region"] = request.Region
	paramMap["ImageName"] = request.ImageName
	paramMap["AccessToken"] = request.AccessToken
	paramMap["Action"] = "SubmitTask"

	if len(request.Cmd) != 0 {
		paramMap["Cmd"] = request.Cmd
	}
	if len(request.OutputDir) != 0 {
		paramMap["OutputDir"] = request.OutputDir
	}
	if len(request.OutputFilename) != 0 {
		paramMap["OutputFileName"] = request.OutputFilename
	}
	if len(request.TaskType) != 0 {
		paramMap["TaskType"] = request.TaskType
	}
	if request.Timeout != 0 {
		paramMap["TimeOut"] = strconv.Itoa(request.Timeout)
	}
	if len(request.TaskName) != 0 {
		paramMap["TaskName"] = request.TaskName
	}

	url := GenerateUrl(TASK_API_URL, paramMap)

	return HttpPost(url, data)
}

type GetTaskResultRequest struct {
	Region      string //required
	TaskId      string //requierd
	AccessToken string //required
}

//获取异步任务的运行结果
/*
获取任务结果失败
{
	"Action" : "GetTaskResultResponse",
	"RetCode" : -1,
	"Message" : “Not Finish”,
	"State" : “Running”,
}

获取任务结果成功
Output-> 返回的tar格式二进制文件
Stdout  任务的标准输出
tmp\result 任务设置的文件输出

*/

func GetTaskResult(request GetTaskResultRequest) (rsp string, err error) {
	paramMap := make(map[string]string)
	paramMap["Region"] = request.Region
	paramMap["TaskId"] = request.TaskId
	paramMap["AccessToken"] = request.AccessToken
	paramMap["Action"] = "GetTaskResult"

	url := GenerateUrl(TASK_API_URL, paramMap)

	return HttpGet(url)
}

File: pythonSdk/sdk/apiInterface.py
# -*- coding: utf-8 -*-

from sdk import apiClient
from config import *
import sys
import json


#API 接口
def GetImageBucketList(Region=REGION, Limit=20, Offset=0, BucketName=None, BucketType=None, OrderBy=None):
	"""
	Args:
		Region String 地域名, 参见地域列表
		BucketName String Docker 镜像仓库名
		BucketType String All： 用户所有可见的仓库 User： 用户个人创建的仓库 Share：第三方公开镜像 默认： All
		OrderBy String Default: 默认排序 
		Limit Integer 默认为 20 
		Offset Integer 默认为 0 

	Returns:
		一个 dict 变量 
		{
		 "Action" :  响应动作,
		 "RetCode" : 执行成功与否， 0 表示成功，
		 "Message" : ,
		 "TotalCount"
		 "BucketSet" : 一个array Docker bucket列表
		 }
	"""
	parameters = { "Action" : "GetImageBucketList", "Region" :  Region, "Limit" : Limit, "Offset" : Offset }
	if BucketType is not None :parameters["BucketType"] = BucketType
	if BucketName is not None : parameters["BucketName"] = BucketName
	if OrderBy is not None : parameters["OrderBy"] = OrderBy
	response   = apiClient._get(COMMON_API_URL, parameters);
	return response


def GetDockerImageList(BucketName, Region=REGION, Limit=20, Offset=0, OrderBy=None):
	"""
	Args:
		Region String 地域名, 参见地域列表
		BuckName String Docker 镜像仓库名称
		OrderBy String Default: 默认排序
		Limit Integer 默认为 0 
		Offset Integer 默认为 20 

	Returns:
		一个 dict 变量 
		{
		 "Action" :  响应动作,
		 "RetCode" : 执行成功与否， 0 表示成功，
		 "Message" : ,
		 "TotalCount"
		 "ImageSet" : 一个array Docker 镜像列表
		 }
	"""
	parameters = { "Action" : "GetDockerImageList", "Region" :  Region, "BucketName" : BucketName, "Limit" : Limit, "Offset" : Offset }
	if OrderBy is not None : parameters["OrderBy"] = OrderBy
	response   = apiClient._get(COMMON_API_URL, parameters);
	return response


def GetAllImages(Region=REGION):
	"""
	Args:
		Region String 地域名, 参见地域列表

	Returns:
		一个 dict 变量 
		{
		 "Action" :  响应动作,
		 "RetCode" : 执行成功与否， 0 表示成功，
		 "Message" : ,
		 "TotalCount"
		 "ImageSet" : 一个array Docker 镜像列表
		 }
	"""
	parameters = { "Action" : "GetAllImages", "Region" :  Region}
	response   = apiClient._get(COMMON_API_URL, parameters);
	return response

def CreateDockerImageBucket(BucketName, Region=REGION):
	"""
	Args:
		Region String 地域名, 参见地域列表
		BuckName String Docker 镜像仓库名称

	Returns:
		一个 dict 变量 
		{
		 "Action" :  响应动作,
		 "RetCode" : 执行成功与否， 0 表示成功，
		 "Message" : ,
		 }
	"""
	parameters = { "Action" : "CreateDockerImageBucket", "Region" :  Region, "BucketName" : BucketName}
	response   = apiClient._get(COMMON_API_URL, parameters);
	return response


def GetTaskList(Region=REGION, Limit=20, Offset=0, TaskId=None, TaskName=None, TaskType= None, State = "None", OrderBy=None, Date=None):
	"""
	Args:
		Region String 地域名, 参见地域列表
		TaskId String Docker 任务 ID 不填返回所有 Id 的任务
		TaskName String 任务名称不填返回所有名称的任务
		TaskType String 同步任务 Sync异步任务 Async 默认为:All
		State String All:所有任务Running： 运行中Success： 成功Fail： 失败 默认为 ALL
		OrderBy String Default: 默认排序
		Limit Integer 默认为 0 
		Offset Integer 默认为 20 
		Date 日期 格式:2016-11-11

	Returns:
		一个 dict 变量 
		{
		 "Action" :  响应动作,
		 "RetCode" : 执行成功与否， 0 表示成功，
		 "Message" : ,
		 "TotalCount"
		 "TaskSet" : 一个array 任务列表
		 }
	"""
	parameters = { "Action" : "GetTaskList", "Region" :  Region, "Limit" : Limit, "Offset" : Offset }
	if TaskId is not None :parameters["TaskId"] = TaskId
	if TaskName is not None : parameters["TaskName"] = TaskName
	if TaskType is not None : parameters["TaskType"] = TaskType
	if State is not None : parameters["State"] = State
	if Date is not None: parameters["Date"] = Date
	response   = apiClient._get(COMMON_API_URL, parameters);
	return response

def GetTaskDetail(TaskId, Region=REGION):
	"""
	Args:
		Region String 地域名, 参见地域列表
		TaskId String Docker 任务 ID 不填返回所有 Id 的任务

	Returns:
		一个 dict 变量 
		{
		"Action" : String 响应动作
		"RetCode" : Integer 执行成功与否， 0 表示成功， 其他值则为错误代码
		"Message" : String 错误原因
		"ImageName" : String 任务使用的镜像名
		"Cmd"  : String 镜像运行参数
		"OutputDir" : String 输出文件目录
		"OutputFileName" : String 输出文件名称
		"TaskType" : String 同步任务 Sync异步任务 Async
		"TaskId" : String 任务 ID
		"TaskName" : String 任务名称
		"Owner" : String 任务创建者
		"State" : String Running： 运行中Success： 成功Fail： 失败
		"CreateTime" : Integer 创建时间， 格式为 Unix 时间戳
		"StartTime" : Integer 任务开始时间， 格式为 Unix时间戳
		"EndTime" : Integer 任务结束时间， 格式为 Unix时间戳
		"Timeout" : Integer 超时时间
		"StdoutBrief" : String 任务标准输出概要
		"StderrBrief" : String 任务标准错误概要
		}
	"""
	parameters = { "Action" : "GetTaskDetail", "Region" :  Region, "TaskId" : TaskId}
	response   = apiClient._get(COMMON_API_URL, parameters);
	return response

def GetAccessToken(Region=REGION, ExpireIn=7200, GrantType=None):
	"""
	Args:
		Region String 地域名, 参见地域列表
		ExpireIn Integer 有效时间默认为 7200 秒
		GrantType String 默认为 Task
	Returns:
		一个 dict 变量 
		{
		 "Action" :  响应动作,
		 "RetCode" : 执行成功与否， 0 表示成功，
		 "Message" : ,
		 "AccessToken" : 授权 Token
		 "ExpireIn" : Token 有效时间
		 }
	"""
	parameters = { "Action" : "GetAccessToken", "Region" :  Region, "ExpireIn" : ExpireIn}
	if GrantType is not None : parameters["GrantType"] = GrantType
	response   = apiClient._get(COMMON_API_URL, parameters);
	return response

def SubmitTask(ImageName, AccessToken, Region=REGION, Cmd=None, OutputDir=None, 
	                    OutputFileName=None, TaskType=None, TimeOut=None, TaskName=None, Data=None):
	"""
	Args:
		Region String 地域名, 参见地域列表
		ImageName String 任务使用的镜像名称
		Cmd String 镜像运行参数
		OutputDir String 输出文件目录
		OutputFileName String 输出文件名称
		TaskType String 同步任务 Sync异步任务 Async 默认为 Sync
		TimeOut Integer 超时时间
		TaskName String 任务名称
		AccessToken String 授权 Token
		Data bytes post数据

	Returns:
		失败: 一个 dict 变量 
		{
		 "Action" :  响应动作,
		 "RetCode" : 执行成功与否， 0 表示成功，
		 "Message" : ,
		 }
		 同步成功: 返回结果二进制数据
		 异步成功: 一个 dict 变量 
		{
		 "Action" :  响应动作,
		 "RetCode" : 执行成功与否， 0 表示成功，
		 "Message" : ,
		 "TaskId"  ： 任务id
		 }
	"""
	parameters = { "Action" : "SubmitTask", "Region" :  Region, "AccessToken" : AccessToken, "ImageName" : ImageName}
	if Cmd is not None : parameters["Cmd"] = Cmd
	if OutputDir is not None : parameters["OutputDir"] = OutputDir
	if OutputFileName is not None : parameters["OutputFileName"] = OutputFileName
	if TaskType is not None : parameters["TaskType"] = TaskType
	if TimeOut is not None : parameters["TimeOut"] = TimeOut
	if TaskName is not None : parameters["TaskName"] = TaskName
	response   = apiClient._post(TASK_API_URL, parameters, Data, Content_Type="application/octet-stream");
	return response

def GetTaskResult(AccessToken, TaskId, Region=REGION):
	"""
	Args:
		Region String 地域名, 参见地域列表
		TaskId String Docker 任务 ID 不填返回所有 Id 的任务
		AccessToken String 授权 Token

	Returns:
		失败 或未运行完成: 一个 dict 变量 
		{
		 "Action" :  响应动作,
		 "RetCode" : 执行成功与否， 0 表示成功，
		 "Message" : ,
		 "State" : 任务运行状态
		 }
		 成功: 返回结果二进制数据
	"""
	parameters = { "Action" : "GetTaskResult", "Region" :  Region, "TaskId" : TaskId, "AccessToken" : AccessToken}
	response   = apiClient._get(TASK_API_URL, parameters);
	return response




